test/neural_networks/test_sampler_qnn.py

```diff
@@ -0,0 +1,279 @@
+            self.assertTrue(isinstance(input_grad, np.ndarray))
+
+    @unittest.skipIf(not _optionals.HAS_SPARSE, "Sparse not available.")
+    @idata(itertools.product(SPARSE, SAMPLERS, INTERPRET_TYPES, BATCH_SIZES))
+    @unpack
+    def test_sampler_qnn(self, sparse: bool, sampler_type, interpret_type, batch_size):
+        """Sampler QNN Test."""
+        qnn = self._get_qnn(sparse, sampler_type, interpret_type)
+        self._verify_qnn(qnn, batch_size)
+
+    @unittest.skipIf(not _optionals.HAS_SPARSE, "Sparse not available.")
+    @idata(itertools.product(SPARSE, INTERPRET_TYPES, BATCH_SIZES))
+    def test_sampler_qnn_gradient(self, config):
+        """Sampler QNN Gradient Test."""
+
+        # get configuration
+        sparse, interpret_id, batch_size = config
+
+        # get QNN
+        qnn = self._get_qnn(sparse, DEFAULT, interpret_id)
+
+        # set input gradients to True
+        qnn.input_gradients = True
+        input_data = np.ones((batch_size, qnn.num_inputs))
+        weights = np.ones(qnn.num_weights)
+        input_grad, weights_grad = qnn.backward(input_data, weights)
+
+        # test input gradients
+        eps = 1e-2
+        for k in range(qnn.num_inputs):
+            delta = np.zeros(input_data.shape)
+            delta[:, k] = eps
+
+            f_1 = qnn.forward(input_data + delta, weights)
+            f_2 = qnn.forward(input_data - delta, weights)
+
+            grad = (f_1 - f_2) / (2 * eps)
+            input_grad_ = input_grad.reshape((batch_size, -1, qnn.num_inputs))[:, :, k].reshape(
+                grad.shape
+            )
+            diff = input_grad_ - grad
+            self.assertAlmostEqual(np.max(np.abs(diff)), 0.0, places=3)
+
+        # test weight gradients
+        eps = 1e-2
+        for k in range(qnn.num_weights):
+            delta = np.zeros(weights.shape)
+            delta[k] = eps
+
+            f_1 = qnn.forward(input_data, weights + delta)
+            f_2 = qnn.forward(input_data, weights - delta)
+
+            grad = (f_1 - f_2) / (2 * eps)
+            weights_grad_ = weights_grad.reshape((batch_size, -1, qnn.num_weights))[
+                :, :, k
+            ].reshape(grad.shape)
+            diff = weights_grad_ - grad
+            self.assertAlmostEqual(np.max(np.abs(diff)), 0.0, places=3)
+
+    def test_circuit_vs_sampler_qnn(self):
+        """Circuit vs Sampler QNN Test. To be removed once CircuitQNN is deprecated"""
+        from qiskit.opflow import Gradient
+        import importlib
+
+        aer = importlib.import_module("qiskit.providers.aer")
+
+        parity = lambda x: f"{x:b}".count("1") % 2
+        output_shape = 2  # this is required in case of a callable with dense output
+
+        qi_qasm = QuantumInstance(
+            aer.AerSimulator(),
+            shots=100,
+            seed_simulator=algorithm_globals.random_seed,
+            seed_transpiler=algorithm_globals.random_seed,
+        )
```

Comment:
Statevector simulator should be used, in this case you should get exact match and the tolerance (now 0.1) could be reduced.